packages/react-router/tests/path.test.ts

```diff
@@ -1,5 +1,50 @@
 import { describe, expect, it } from 'vitest'
-import { exactPathTest, removeTrailingSlash } from '../src/path'
+import { exactPathTest, removeBasepath, removeTrailingSlash } from '../src/path'
+
+describe('removeBasepath', () => {
+  ;[
+    {
+      name: '`/` should leave pathname as-is',
+      basepath: '/',
+      pathname: '/path',
+      expected: '/path',
+    },
+    {
+      name: 'should return empty string if basepath is the same as pathname',
+      basepath: '/path',
+      pathname: '/path',
+      expected: '',
+    },
+    {
+      name: 'should remove basepath from the beginning of the pathname',
+      basepath: '/app',
+      pathname: '/app/path/app',
+      expected: '/path/app',
+    },
+    {
+      name: 'should remove multisegment basepath from the beginning of the pathname',
+      basepath: '/app/new',
+      pathname: '/app/new/path/app/new',
+      expected: '/path/app/new',
+    },
+    {
+      name: 'should remove basepath only in case it matches segments completely',
+      basepath: '/app',
+      pathname: '/application',
+      expected: '/application',
+    },
+    {
+      name: 'should remove multisegment basepath only in case it matches segments completely',
+      basepath: '/app/new',
+      pathname: '/app/new-application',
+      expected: '/app/new-application',
+    },
+  ].forEach(({ name, basepath, pathname, expected }) => {
```

Comment:
please use the `describe.each` syntax: https://vitest.dev/api/#describe-each

Reply:
According vitest's docs it will be a misuse of `each` since it is designed for `more than one test that depends on the same data` which is not true in this case since written is simply a table-test.
But okay.

Reply:
then `test.each`? https://vitest.dev/api/#test-each

Reply:
This will do)